### PR TITLE
Jakarta Pages 3.0

### PR DESCRIPTION
**Pull request by pnicolucci**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Generated IP Log
   - [x] Email to PMC
   - [x] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/servlet/jsp/jakarta.servlet.jsp-api/3.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-pages-tck-3.0.0.zip
- [x] The URL of the compatibility certification request issue:
       eclipse-ee4j/jsp-api#190
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by kazumura** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

File: pages/3.0/_index.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Jakarta Server Pages 3.0 (milestone)"
+title: "Jakarta Server Pages 3.0"
 date: 2020-06-13
```
Comment (kwsutter) [inline pages/3.0/_index.md:3]: This date will need to be changed at some point.  When you get closer to a ballot, you can put in a more reasonable date.

File: pages/3.0/_index.md [hunk 2]
```diff
@@ -9,17 +9,17 @@ into a Jakarta Servlet.
 
 * [Jakarta Server Pages 3.0 Release Record](https://projects.eclipse.org/projects/ee4j.jsp/releases/3.0.0)
```
Comment (kwsutter) [inline pages/3.0/_index.md:10]: The release date in this release record will need to be adjusted.  Maybe push for late October.  An Eclipse release review will also need to be initiated at your earliest convenience.  An IP log generated with notes to EMO and PMC.  Should be pretty boilerplate.

File: pages/3.0/_index.md (hunk 1 above)
Comment (pnicolucci) [reply pages/3.0/_index.md]: Updated date to today, we can update again if necessary.

File: pages/3.0/_index.md (hunk 2 above)
Comment (pnicolucci) [reply pages/3.0/_index.md:10]: Changed release date to Oct 26, 2020 for now.

**Comment by kwsutter** [comment]
The Javadocs have a version of "Jakarta Server Pages API v3.0.0-SNAPSHOT" on its banner page.

**Comment by pnicolucci** [comment]
> The Javadocs have a version of "Jakarta Server Pages API v3.0.0-SNAPSHOT" on its banner page.

Updated to "Jakarta Server Pages API v3.0.0"

File: pages/3.0/_index.md (hunk 2 above)
Comment (kazumura) [reply pages/3.0/_index.md:10]: The link must be "https://projects.eclipse.org/projects/ee4j.jsp/releases/3.0.0/plan" but not "https://projects.eclipse.org/projects/ee4j.jsp/releases/3.0.0".

File: pages/3.0/_index.md [hunk 3]
```diff
@@ -9,17 +9,17 @@ into a Jakarta Servlet.
...
-* [Jakarta Server Pages 3.0 Specification Document](./server-pages-spec-3.0-RC2.html) (HTML)
+* [Jakarta Server Pages 3.0 Specification Document](./jakarta-server-pages-spec-3.0.pdf) (PDF)
+* [Jakarta Server Pages 3.0 Specification Document](./jakarta-server-pages-spec-3.0.html) (HTML)
 * [Jakarta Server Pages 3.0 Javadoc](./apidocs)
-* [Jakarta Server Pages 3.0 TCK](http://download.eclipse.org/ee4j/jakartaee-tck/master/nightly/pages-tck-3.0.0.zip)([sig](),[sha](),[pub]())
+* [Jakarta Server Pages 3.0 TCK](https://download.eclipse.org/jakartaee/pages/3.0/jakarta-pages-tck-3.0.0.zip)([sig](https://download.eclipse.org/jakartaee/pages/3.0/jakarta-pages-tck-3.0.0.zip.sig),[sha](https://download.eclipse.org/jakartaee/pages/3.0/jakarta-pages-tck-3.0.0.zip.sha256),[pub](https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub))
 * Maven coordinates
-  * [jakarta.pages:jakarta.pages-api:jar:3.0.0](https://repo1.maven.org/maven2/jakarta/servlet/jsp/jakarta.servlet.jsp-api/3.0.0-RC2/)
+  * [jakarta.pages:jakarta.pages-api:jar:3.0.0](https://repo1.maven.org/maven2/jakarta/servlet/jsp/jakarta.servlet.jsp-api/3.0.0/)
```
Comment (kazumura) [inline pages/3.0/_index.md:17]: The link of API jar must be "https://search.maven.org/artifact/jakarta.servlet.jsp/jakarta.servlet.jsp-api/3.0.0/jar".

File: pages/3.0/_index.md (hunk 2 above)
Comment (kwsutter) [reply pages/3.0/_index.md:10]: Either link is fine for the release record.

File: pages/3.0/_index.md (hunk 3 above)
Comment (kwsutter) [reply pages/3.0/_index.md:17]: Referencing the jar directory would help clarify....

Comment (pnicolucci) [reply pages/3.0/_index.md:17]: ok so just to clarify @kwsutter  we should use this link? [](https://repo1.maven.org/maven2/jakarta/servlet/jsp/jakarta.servlet.jsp-api/3.0.0/jar) ? If so I can get this change pushed up.

**Comment by kwsutter** [comment]
Working with @pnicolucci to get this Spec PR ready.  

- There are a couple of updates to the Specification itself with the use of the JSP abbreviation. Paul is taking care of those updates.
- Paul is also taking care of the other minor items identified by @kazumura via earlier reviews.
- The latest TCK run is 100% with GF and the JSP TCK 3.0.0 bucket.
- The staged APIs are about to run out of runway (60 day limit)...  Paul is checking with the JSP team to see how these artifacts can be refreshed in staging.  Should not be a big deal.
- Still need to find a location for the public TCK results.  Any ideas would be appreciated.
- Need to create a CR.  Once we get a TCK Results page, this should be easy.

Once these things are done, then we can ask @kazumura for another review and, hopefully, be ready for a ballot.

File: pages/3.0/_index.md [hunk 4]
```diff
@@ -9,17 +9,17 @@ into a Jakarta Servlet.
...
 
 
 # Compatible Implementations
 
-* [Apache Tomcat 10.0.0-M6](https://tomcat.apache.org/download-10.cgi)
+* [Apache Tomcat 10.0.0-M8](https://tomcat.apache.org/download-10.cgi)
```
Comment (kwsutter) [inline pages/3.0/_index.md:22]: @pnicolucci Can you update this to also point at the Glassfish RC 1 deliverable?  
https://download.eclipse.org/ee4j/glassfish/glassfish-6.0.0-RC1.zip

**Comment by arjantijms** [comment]
> The staged APIs are about to run out of runway (60 day limit)... Paul is checking with the JSP team to see how these artifacts can be refreshed in staging. Should not be a big deal.

I quickly refreshed the APIs. It's no big deal indeed. 

(did notice an error in our script - dry run doesn't guard against auto dropping the staging repo)

**Comment by arjantijms** [comment]
@pnicolucci on request I filed the CR here: https://github.com/eclipse-ee4j/jsp-api/issues/190

Can you update the description with this? Thx!

**Comment by kazumura** [comment]
@pnicolucci 
In order to proceed to the ballot, please fix the followings:

* change API jar link in _index.md.
* add GlassFish to CI in _index.md.

File: pages/3.0/_index.md (hunk 3 above)
Comment (pnicolucci) [reply pages/3.0/_index.md]: updated link in latest commit

File: pages/3.0/_index.md (hunk 4 above)
Comment (pnicolucci) [reply pages/3.0/_index.md]: updated in latest commit

**Comment by pnicolucci** [comment]
@kwsutter @kazumura , I've refreshed the api and spec documents as well as made the requested changes to the `_index.md`. Let me know if there are any other  changes you would like made.

**Comment by pnicolucci** [comment]
Just updated `_index.md` to include the Jakarta Server Pages 3.0.0 Impl as a CI.

**Review by kwsutter — APPROVED** [review]
LGTM.  Thanks, @pnicolucci and @arjantijms, for doing the final touches on this PR.  We'll leave this open for @kazumura's final review.  If he approves, then he can start the ballot.  Thanks again!

**Review by kazumura — APPROVED** [review]
LGM.
Thanks, @pnicolucci , @arjantijms .

**Comment by arjantijms** [comment]
Thanks @kazumura

**Comment by kazumura** [comment]
Ballot thread
https://www.eclipse.org/lists/jakarta.ee-spec/msg01163.html

**Comment by kwsutter** [comment]
A polite nudge for @kazumura to complete the final checklist for Server Pages... Thanks!

* On ballot completion, the specification committee mentor:
 - [ ] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>


 ----
 - [ ] promotes api staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] go through the merged jakarta.ee specification website page to verify all the links are valid.
 - [ ] approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] merge any final release branch as appropriate for the branch management for the project.

**Comment by arjantijms** [comment]
Result of ballot: https://eclipse.org/lists/jakarta.ee-spec/msg01303.html

**Comment by arjantijms** [comment]
Release bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=568078

**Pull request merged**